packages/web/src/utils/pool-utils.spec.ts

```
@@ -0,0 +1,133 @@
+
+        expect(lexicalOrder).toBe(sortOrder);
+      }
+    }
+  });
+
+  // When paths are equal, `<=` returns true, but sortTokenPaths[0] === pathA is also true.
+  // This is fine. The real concern is whether checkGnotPath could introduce a case
+  // where GNOT -> wrappedPath changes the ordering result.
+  it("should handle equal paths correctly - both <= and sort return true", () => {
+    const samePath = "gno.land/r/gnoswap/gns";
+    expect(samePath <= samePath).toBe(true);
```

Biome (2.4.4)

[error] Comparing to itself is potentially pointless.

(lint/suspicious/noSelfCompare)